mteb/tasks/Retrieval/ara/SadeemKeywordRetrieval.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import annotations
+
+import datasets
+
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+from ....abstasks.AbsTaskRetrieval import AbsTaskRetrieval
+
+
+class SadeemKeywordRetrieval(AbsTaskRetrieval):
+    _EVAL_SPLIT = "test"
+
+    metadata = TaskMetadata(
+        name="SadeemKeywordRetrieval",
+        dataset={
+            "path": "sadeem-ai/sadeem-ar-eval-retrieval-keywords",
+            "revision": "b76ba10cc7c6a41f9bd69e7dfa57e5ae17ddd761",
+            "name": "test",
+        },
+        reference="https://huggingface.co/datasets/sadeem-ai/sadeem-ar-eval-retrieval-keywords",
+        description="SadeemKeyword: A Benchmark Data Set for Community Keyword-Retrieval Research",
+        type="Retrieval",
+        category="s2p",
+        eval_splits=[_EVAL_SPLIT],
+        eval_langs=["ara-Arab"],
+        main_score="ndcg_at_10",
+        date=None,
+        form=["written"],
+        domains=["Blog"],
+        task_subtypes=None,
+        license=None,
+        socioeconomic_status=None,
+        annotations_creators=None,
+        dialect=None,
+        text_creation=None,
+        bibtex_citation=None,
+        n_samples={_EVAL_SPLIT: 7179},
+        avg_character_length={_EVAL_SPLIT: 500.0},
```

Comment:
The reason why the test fails is because the metadata is not filled which it should be.

date is the time that the text were written (e.g. scraped from twitter from 2001-2020)
task_subtype I would put Keyword Retrieval and add it to the list of allowed subtypes
license is required
socioeconomic status is the social status of the text writers (e.g. high for lawyers).
dialect should be an empty list if there are no dialects

You can read more about these on the TaskMetadata object